.github/workflows/build-user-config.yml

```diff
@@ -0,0 +1,120 @@
+name: Reusable user config build
+
+on: 
+  workflow_call:
+    inputs:
+      build_matrix_path:
+        description: 'Path to the build matrix file'
+        default: 'build.yaml'
+        required: false
+        type: string
+      config_path:
+        description: 'Path to the config directory'
+        default: 'config'
+        required: false
+        type: string
+      fallback_binary:
+        description: 'Fallback binary format, if no *.uf2 file was built'
+        default: 'bin'
+        required: false
+        type: string
+
+jobs:
+  matrix:
+    runs-on: ubuntu-latest
+    name: Fetch Build Keyboards
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install yaml2json
+        run: python3 -m pip install remarshal
+
+      - id: set-matrix
+        name: Fetch Build Matrix
+        run: |
+          matrix=$(yaml2json ${{ inputs.build_matrix_path }} | jq -c .)
+          yaml2json ${{ inputs.build_matrix_path }}
+          echo "::set-output name=matrix::${matrix}"
+
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: zmkfirmware/zmk-build-arm:stable
+    needs: matrix
+    name: Build
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.matrix.outputs.matrix)}}
+    steps:
+      - name: Prepare variables
+        id: variables
+        run: |
+          if [ -n "${{ matrix.shield }}" ]; then
+            EXTRA_CMAKE_ARGS="-DSHIELD=${{ matrix.shield }}"
+            ARTIFACT_NAME="${{ matrix.shield }}-${{ matrix.board }}-zmk"
+            DISPLAY_NAME="${{ matrix.shield }} - ${{ matrix.board }}"
+          else
+            EXTRA_CMAKE_ARGS=
+            DISPLAY_NAME="${{ matrix.board }}"
+            ARTIFACT_NAME="${{ matrix.board }}-zmk"
+          fi
+          echo ::set-output name=extra-cmake-args::${EXTRA_CMAKE_ARGS}
+          echo ::set-output name=artifact-name::${ARTIFACT_NAME}
+          echo ::set-output name=display-name::${DISPLAY_NAME}
+          echo ::set-output name=zephyr-version::${ZEPHYR_VERSION}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Cache west modules
+        uses: actions/cache@v3.0.1
+        env:
+          cache-name: cache-zephyr-${{ steps.variables.outputs.zephyr-version }}-modules
+        with:
+          path: |
+            modules/
+            tools/
+            zephyr/
+            bootloader/
+            zmk/
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('${{ inputs.config_path }}/west.yml') }}
```

Comment:
The use of the `inputs` here isn't working as expected on pushes. It ends up not finding the `west.yml` file at all in that scenario. See https://github.com/petejohanson/testing-3-0-update-zmk-config/runs/5809977931?check_suite_focus=true

Reply:
Yes, that is a bit strange. I will investigate.

Reply:
I really like the idea of this change. Wonder if we could skip the input parameters for workflow run for now, to get this in, then iterate?

Reply:
Perhaps `hashFiles('**/west.yml', '**/build.yaml')` will be sufficient for unique hashes from user's config folder at this point in time?

Reply:
Yes, I think that has the same effect, since you hash the same files.

Reply:
Maybe GH actions don't support interpolating inside single quote strings or nested interpolations? Using globs like that seems like it should work well though.

Reply:
I found references of GH actions that do path concatenation this way with success, so I don't think it's a problem. However, it could also be changed to something like `join( <array of  path components>, '/' )`. In any case, globs are probably most general and catch all matching files too.